apps/condo/domains/user/schema/ConfirmPhoneAction.js

```diff
@@ -0,0 +1,245 @@
+                const smsCode = generateSmsCode(phone)
+                const smsCodeRequestedAt = new Date(now).toISOString()
+                const smsCodeExpiresAt = new Date(now + SMS_CODE_TTL * 1000).toISOString()
+                const variables = { 
+                    dv,
+                    sender,
+                    phone,
+                    smsCode,
+                    token,
+                    smsCodeRequestedAt,
+                    smsCodeExpiresAt,
+                    requestedAt,
+                    expiresAt,
+                }
+                await ConfirmPhoneActionGQL.create(context.createContext({ skipAccessControl: true }), variables)
+                const lang = COUNTRIES[RUSSIA_COUNTRY].locale
+                await sendMessage(context, {
+                    lang,
+                    to: { phone },
+                    type: SMS_VERIFY_CODE_MESSAGE_TYPE,
+                    meta: {
+                        dv: 1,
+                        smsCode,                        
+                    },
+                    sender: sender,
+                })
+                return token
+            },
+        },
+        {
+            access: true,
+            schema: 'confirmPhoneActionResendSms(token: String!, sender: JSON!): String',
+            resolver: async (parent, args, context, info, extra) => {
+                const { token, sender } = args
+                const now = extra.extraNow || Date.now()
+                const actions = await ConfirmPhoneActionGQL.getAll(context.createContext({ skipAccessControl: true }), { 
+                    token,
+                    expiresAt_gte: new Date(now).toISOString(),
+                    completedAt: null,
+                })
+                if (isEmpty(actions)) {
+                    throw new Error('[error]: Unable to find confirm phone action by token')
+                }
+                const { id, phone } = actions[0]
+                const newSmsCode = generateSmsCode(phone)
+                await ConfirmPhoneActionGQL.update(context.createContext({ skipAccessControl: true }), id, { 
+                    smsCode: newSmsCode,
+                    smsCodeExpiresAt:  new Date(now + SMS_CODE_TTL * 1000).toISOString(),
+                    smsCodeRequestedAt: new Date(now).toISOString(),
+                })
+                const lang = COUNTRIES[RUSSIA_COUNTRY].locale        
+                await sendMessage(context, {
+                    lang,
+                    to: { phone },
+                    type: SMS_VERIFY_CODE_MESSAGE_TYPE,
+                    meta: {
+                        dv: 1,
+                        smsCode: newSmsCode,                        
+                    },
+                    sender: sender,
+                })
+                return 'ok'
+            },
+        },
+        {
+            access: true,
+            schema: 'completeConfirmPhoneAction(token: String!, smsCode: Int!): String',
```

Comment:
it's better to use `completeConfirmPhoneAction(data: CompleteConfirmPhoneActionInput!): CompleteConfirmPhoneActionOutput` to extend the return data